Created RadioButton component

SofiiaYevush opened the pull request:
Created RadioButton component that meets the design system standards. #2703 

**Examples of RadioButton in Storybook:**

1. ![image](https://github.com/user-attachments/assets/0e41f42f-ef8a-42d0-9ef2-bb1336a6c8c8) 
3. ![image](https://github.com/user-attachments/assets/ad8200a3-3f39-42ba-9ac5-426aded0c7ce) 
3. ![image](https://github.com/user-attachments/assets/3cb7f410-335a-43f9-9d32-98bc94eb36bd)
[ img1, img2, img3: Relevant size, color, label position, checked:true ]



4. ![image](https://github.com/user-attachments/assets/6668caba-2379-480b-ac72-2db95e31f9af)
[ img4: checked:false ]



5. ![image](https://github.com/user-attachments/assets/e45c4e2a-e7fd-42bb-bd16-67c076e40f1b)
[ img5: disabled:true ]



6. ![image](https://github.com/user-attachments/assets/8ca1aa89-397d-426d-9f19-56e8d8985698)
[ img6: loading:true ]

SofiiaYevush committed:
Created RadioButton component

Mav-Ivan reviewed (changes requested):
move component to src/design-system/components
move stories to src/design-system/stories
move test to tests/unit/design-system
Also use aliases `~scss-components `, `~scss`

SofiiaYevush committed:
Merge branch 'develop' of https://github.com/ita-social-projects/SpaceToStudy-Client into feature/2703/creation-radio-button-component

SofiiaYevush committed:
Moving files into design-system folders and changing directions

SofiiaYevush commented:
> move component to src/design-system/components move stories to src/design-system/stories move test to tests/unit/design-system Also use aliases `~scss-components `, `~scss`

Done. 
Moved: `RadioButton.tsx, RadioButton.scss` to `src/design-system/components/radio-button`;
`RadioButton.stories.tsx` to `src/design-system/stories`;
`RadioButton.spec.jsx` to `tests/unit/design-system/components/radio-button`.
Added `~scss-components `, `~scss`.

SofiiaYevush committed:
Deleting commented code in files

Mav-Ivan commented on src/design-system/components/radio-button/RadioButton.tsx:
```diff
@@ -0,0 +1,85 @@
+  ) => {
+    const isDisabled = disabled || loading
+
+    const loader = loading ? (
```
add loading && (...

SofiiaYevush replied:
> add loading && (...

Done. Changed logic for using loading.

SofiiaYevush committed:
Refactored RadioButton loader logic
